 Fix WaveGenerator memory leak

Releasing memory with ToneGenerator::stopTone() depends on
ToneGenerator::mState that can cause memory leak in case
mState is TONE_INIT.

Remove dependency to mState by adding ToneGenerator::clearWaveGens()
explicit call in ToneGenerator destructor.

diff --git a/media/libmedia/ToneGenerator.cpp b/media/libmedia/ToneGenerator.cpp
index 44c7edc..b108078 100644
--- a/media/libmedia/ToneGenerator.cpp
+++ b/media/libmedia/ToneGenerator.cpp
@@ -865,6 +865,7 @@ ToneGenerator::~ToneGenerator() {
         ALOGV("Delete Track: %p", mpAudioTrack.get());
         mpAudioTrack.clear();
     }
+    clearWaveGens();
 }
 
 ////////////////////////////////////////////////////////////////////////////////
